.devops/templates/pr-ci.yml

```
@@ -0,0 +1,157 @@
+
+  - template: variables.yml
+
+jobs:
+  - job: BuildTestLint
+    workspace:
+      clean: all
+    steps:
+      - template: tools.yml
+
+      - task: Bash@3
+        inputs:
+          inputs:
+          filePath: yarn-ci.sh
+        displayName: yarn
+
+      - script: |
+          yarn checkchange
+        displayName: check change
+
+      ## manually call this since lage is now used and it calls lint per package, not per repo
+      - script: |
+          yarn prelint &&
+          yarn prettier $(targetBranch) --check
+        displayName: do syncpack, lint-files, and prettier checks
+
+      ## Danger.js checks for Fluent UI N*
+      - ${{ if eq(parameters.isPR, true) }}:
+          - script: |
+              yarn danger ci
+            displayName: danger
+            env:
+              DANGER_GITHUB_API_TOKEN: $(DANGER_GITHUB_API_TOKEN)
+
+      - script: |
+          yarn buildci --no-cache $(sinceArg)
+        displayName: build, test, lint
+
+      - template: cleanup.yml
+        parameters:
+          checkForChangedFiles: false
+
+  - job: Deploy
+    workspace:
+      clean: all
+    variables:
+      - template: variables.yml
+    steps:
+      - template: tools.yml
+
+      - task: Bash@3
+        inputs:
+          filePath: yarn-ci.sh
+        displayName: yarn
+
+      - script: |
+          yarn bundle --no-cache $(sinceArg)
+        displayName: bundle
+
+      ## This runs regardless of scope, the app will adapt to the scope as well
+      - script: |
+          yarn workspace @fluentui/pr-deploy-site generate:site
+        displayName: generate PR Deploy Site
+
+      - publish: $(Build.ArtifactStagingDirectory)
+        artifact: Build-PR-$(Build.BuildNumber)
+
+      - task: AzureUpload@2
+        displayName: Upload PR deploy site
+        inputs:
+          azureSubscription: $(azureSubscription)
+          BlobPrefix: $(deployBasePath)
+          ContainerName: '$web'
+          SourcePath: 'apps/pr-deploy-site/dist'
+          storage: fabricweb
+
+      - task: GithubPRStatus@0
+        displayName: 'Update PR deploy site github status'
+        inputs:
+          githubOwner: microsoft
+          githubRepo: fluentui
+          githubContext: 'Pull request demo site'
+          githubDescription: 'Click "Details" to go to the deployed demo site for this pull request'
+          githubTargetLink: 'http://fabricweb.z5.web.core.windows.net/pr-deploy-site/$(Build.SourceBranch)/'
+
+      - ${{ if eq(parameters.isPR, false) }}:
+          - task: AzureUpload@2
+            displayName: Upload demo images
+            inputs:
+              azureSubscription: $(azureSubscription)
+              BlobPrefix: 'assets'
+              CacheControl: 'public, max-age=600000'
+              ContainerName: 'fabric-website' # this container has a CDN
+              SourcePath: 'packages/fluentui/docs/src/public'
+              storage: fabricweb
+
+      - template: cleanup.yml
+
+  - job: ScreenerFluent
```

```suggestion
  - job: ScreenerNorthstar
```

nit: As we touched this file, can we improve naming of this job? 🐱

---

done